services/app-web/src/hooks/useScrollToPageTop.ts

```diff
@@ -0,0 +1,11 @@
+import { useLayoutEffect } from 'react'
+import { useLocation } from 'react-router'
+
+// Scroll to top of page on load, used on every page change
+export const useScrollToPageTop = (): void => {
+    const { pathname } = useLocation()
+
+    useLayoutEffect(() => {
+        window.scrollTo(0, 0)
+    }, [pathname])
+}
```

Comment:
I like how you made this into a hook! Worked well when I tested this.

I'm interested to see how we tackle this scroll and the `FieldPreserveScrollPosition`, since they both would be firing.

Reply:
Yes! I thought about this a bit. Let write out some thoughts here.
- The useScrollToPageTop effect will only fire _on initial mount_ with the route change. 
- The effect in FieldPreserveScrollPosition fires _if the formik field value changes after initial mount._ I  believe the only way that formik values could change after the initial mount is from some user interaction like clicking or typing in the field.

I don't think these should conflict. I tested this with and already filled out and saved contract details page (e.g. there is data that is not undefined/empty in there for the contract type field) and slow throttling to slow the api request. It still seems to work fine (only goes scrolls to the top on load - it doesn't try to jump down at all when the api returns the data for contract type). 

What do you think? Do you feel comfortable merging this?

Reply:
>I'm interested to see how we tackle this scroll and the FieldPreserveScrollPosition, since they both would be firing.

Initially I mentioned this because of what I saw going back to `Contract Details` page.
>Looks great! There was an issue the first round of testing, where going back to the `Contract Details` page would scroll to the fields, this is no longer happening. I wonder if changing this to `useLayoutEffect` from `useEffect` fix that because, I think `useEffect` fires after `useLayoutEffect`.

Otherwise everything looks awesome!